CARDS-2540: Add ability to submit questionnaire responses starting with a certain day relative to the appointment, regardless of completion status

marta- opened the pull request:
Specs: https://phenotips.atlassian.net/browse/CARDS-2540

The implementation differs slightly from the specs: instead of enabling incomplete submission 24h before, it can be enabled the day of the visit, or the day before the visit, or any number of days before or after the visit. Incomplete submission becomes enabled at midnight.

To test:
* start in `prems` or `proms`
* Administration / Patient Access -> change the incomplete submission setting according to helper text
* Create visits with surveys
   * test that visits occurring earlier than `<setting>` days before visit are (still) required to be complete in order to be submitted
   * test that visits occurring later than `<setting>` days before visit are NOT required to be complete in order to be submitted

marta- committed:
CARDS-2540: Add ability to submit questionnaire responses for a certain time  prior to scheduled appointments regardless of completion status

marta- committed:
CARDS-2540: Add ability to submit questionnaire responses for a certain time relative to scheduled appointments regardless of completion status

Extended the functionality to also allow enabling incomplete submissions a certain time after the appointment

marta- commented:
Addressing comment from @sdumitriu's in 395e812 : allow for enabling incomplete submission after the visit as well.

sashaandjic reviewed (approved):
tested

It works well , however this screen may confuse the patient:
![image](https://github.com/data-team-uhn/cards/assets/23243244/8d81ddd2-f6b0-44c4-ab99-c13f0868fd67)

I would skip this screen and allow patient to proceed to the next page, but once the submit button is pressed I would let them know that they are submitting incomplete answers and give them opportunity to either proceed or cancel the submission and go back to the complete the surveys. 
Or more detailed warning message: "Your answers are not complete. Do you want to submit them incomplete, or go back and complete the surveys?"

Basically, I would like to change the last 2 steps for incomplete answers:
- allow pressing the SUBMIT button
- if things are not perfect, ask user if they want to proceed or not

acrowthe reviewed (approved):
Code looks good - not tested

marta- committed:
CARDS-2540: Add ability to submit questionnaire responses for a certain time relative to scheduled appointments regardless of completion status
Better presentation of Update and Proceed buttons on the incomplete survey screen

sashaandjic reviewed (approved):
tested

![image](https://github.com/data-team-uhn/cards/assets/23243244/82e4ab9c-d6aa-40d8-8dce-ff7531335659)
